src/ts/background/store/api-context/reducer.ts

```diff
@@ -0,0 +1,35 @@
+import ApiPromise from '@polkadot/api/promise'
+import { ProviderInterface } from '@polkadot/rpc-provider/types'
+import { AnyAction, Reducer } from 'redux'
+import { ACTION_TYPES } from './actions'
+import { REQUEST, SUCCESS, FAILURE } from '../util'
+
+export interface IApiContext {
+  apiReady: boolean,
+  provider?: ProviderInterface
+  api?: ApiPromise
+}
+
+const initialState: IApiContext = {
+  apiReady: false
+}
+
+export const apiContext: Reducer<IApiContext, AnyAction> = (state = initialState, action) => {
+  switch (action.type) {
+    case REQUEST(ACTION_TYPES.CONNECT):
+      // disconnect current network if any
+      state.provider && state.provider.isConnected() && state.provider.disconnect()
```

Comment:
It would be better to do the actual disconnection in the action function while only updating state in reducer.

Reply:
According to https://redux.js.org/basics/reducers and https://redux.js.org/basics/actions 
```
Actions are payloads of information that send data from your application to your store. They are the only source of information for the store.
```
```
Reducers specify how the application's state changes in response to actions sent to the store. Remember that **actions only describe what happened, but don't describe how the application's state changes**.
```
Disconnection is more state change than information, I think, which is why it's put in reducer.

Reply:
> The reducer is a pure function that takes the previous state and an action, and returns the next state.
> Things you should never do inside a reducer: 
> - Mutate its arguments;
> - Perform side effects like API calls and routing transitions;
> - Call non-pure functions, e.g. Date.now() or Math.random().

For error handling, i think it may be easier to handle connection outside reducer.

Reply:
test

Reply:
test

Reply:
> The reducer is a pure function that takes the previous state and an action, and returns the next state.
> Things you should never do inside a reducer: 
> - Mutate its arguments;
> - Perform side effects like API calls and routing transitions;
> - Call non-pure functions, e.g. Date.now() or Math.random().

For error handling, i think it may be easier to handle connection outside reducer.